### PR TITLE
Added panel handler to run actions from vpn popup

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/22874
Resolves https://github.com/brave/brave-browser/issues/22711
Resolves https://github.com/brave/brave-browser/issues/22715

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Open popup with clean profile and click 'already purchased' link
- Check a new tab opened

### CODE REVIEW

**Review by bsclifton — APPROVED** [review]
++

**Comment by petemill** [comment]
Please check with security team that they are ok putting an extension API in chrome-untrusted. As far as I can see, this defeats the point of using chrome-untrusted. If we deem the code in the frame is not trustworthy and could be susceptible to being abused, then we would want to limit the abuse surface - and chrome.tabs has some things we might not want to be abused (e.g. opening a tab to a phishing site).

**Review by bsclifton — CHANGES_REQUESTED** [review]
Actually; let's do with handlers as we discussed 😄

**Review by petemill — CHANGES_REQUESTED** [review]
Marking as "Request changes" until we have a security review or take a different approach. I think a preferred approach would be to add Mojom methods for UI actions, e.g. `VisitSupportPage` would open the necessary URL in a new tab. That would be better for cross-platform too.

**Comment by petemill** [comment]
brave_vpn.mojom's ServiceHandler also seems like a good cross-platform place to put the handler's for these UI actions.

**Comment by bsclifton** [comment]
This looks great! There are a few other places where chrome.tabs.create was used (doh; my bad). I think we can effectively remove this panel from that whitelist after this change is complete (no reason to keep `brave://vpn-panel.top-chrome/`)

Here are the places:
- `src\brave\components\brave_vpn\resources\panel\components\contact-support\index.tsx`
- `src\brave\components\brave_vpn\resources\panel\components\error-subscription-failed-panel\index.tsx`
- `src\brave\components\brave_vpn\resources\panel\components\settings-panel\index.tsx`

File: components/brave_vpn/mojom/brave_vpn.mojom [hunk 1]
```diff
@@ -20,6 +20,9 @@ interface PanelHandler {
 
   // Notify the backend that the dialog should be closed.
   CloseUI();
+
+  // Run purchase action.
+  RunPurchaseFlow(string action);
```
Comment (petemill) [inline components/brave_vpn/mojom/brave_vpn.mojom:25]: You think `PanelHandler` is a better place than `ServiceHandler`?

File: common/extensions/api/_api_features.json [hunk 2]
```diff
@@ -73,8 +73,7 @@
       "chrome://tab-strip/*",
       "chrome://wallet-panel.top-chrome/*",
       "chrome://wallet/*",
-      "chrome://brave-shields.top-chrome/*",
-      "chrome://vpn-panel.top-chrome/*"
+      "chrome://brave-shields.top-chrome/*"
```
Comment (bsclifton) [inline common/extensions/api/_api_features.json:76]: This is the one where I think we can just remove it altogether. No extension API access needed anymore 😄

File: components/brave_vpn/mojom/brave_vpn.mojom (hunk 1 above)
Comment (spylogsster) [reply components/brave_vpn/mojom/brave_vpn.mojom:25]: yes, we need access to `chrome/browser/ui` and `VPNPanelHandler` has it. `ServiceHandler` is on component level. Also that seems just a click handlers, not the service logic

**Comment by spylogsster** [comment]
> * resources\panel\components\settings-panel\index.tsx

done

File: common/extensions/api/_api_features.json (hunk 2 above)
Comment (bsclifton) [reply common/extensions/api/_api_features.json:76]: Fixed! perfect, thanks 😄

**Review by bsclifton — APPROVED** [review]
Works beautifully 😄👍

File: components/brave_vpn/mojom/brave_vpn.mojom (hunk 1 above)
Comment (petemill) [reply components/brave_vpn/mojom/brave_vpn.mojom]: I guess it's fine and if android needs similar we can think about it or duplicate it. But it's a click handler just as much as any other buttons which result in calls to service handler.

**Review by petemill — APPROVED** [review]

**Pull request merged**